src/lib/modules/blockchain_connector/provider.js

```diff
@@ -16,9 +20,31 @@ class Provider {
 
   startWeb3Provider(callback) {
     const self = this;
+    const accountsConfig = self.accountsConfig[0];
 
     if (this.type === 'rpc') {
       self.provider = new this.web3.providers.HttpProvider(self.web3Endpoint);
+    } else if (this.type === 'ledger') {
+        if (accountsConfig.HardwareWallet === 'Ledger') {
+
+          const rpcUrl = accountsConfig.HardwareWallet.RpcUrl;
+          const engine = new ProviderEngine();
+          const getTransport = () => TransportNodeHid.open(accountsConfig.LedgerWalletHardwareAddress);
+          const ledger = createLedgerSubprovider(getTransport, {
+            networkId: accountsConfig.NetworkId,  // refer to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+            accountsLength: accountsConfig.NoOfLedgerAccountsToLoad // number of accounts to load from the device
+          });
+
+          engine.addProvider(ledger); 
+          engine.addProvider(new FetchSubprovider({ rpcUrl }));
+          engine.start();
```

Comment:
Could you put the `start` after the event listeners below, just in case the engine fails immediately.